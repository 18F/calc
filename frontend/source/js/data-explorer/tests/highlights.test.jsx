/* global expect, describe, it */
import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Highlights } from '../components/highlights.jsx';

function setup() {
  const props = {
    stdDeviation: 1.1,
    avgPrice: 2.1,
    proposedPrice: 5,
  };

  const wrapper = render(<Highlights {...props} />);

  return {
    props,
    wrapper,
  };
}

describe('<Highlights>', () => {
  it('renders correctly', () => {
    const { wrapper } = setup();

    expect(wrapper.find('.sd-highlight').first().text()).toBe('$1');
    expect(wrapper.find('.avg-price-highlight').text()).toBe('$2');
    expect(wrapper.find('.sd-highlight').last().text()).toBe('$3');
    expect(wrapper.find('.proposed-price-highlight').text()).toBe('$5');
  });

  it('matches snapshot', () => {
    const { wrapper } = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
