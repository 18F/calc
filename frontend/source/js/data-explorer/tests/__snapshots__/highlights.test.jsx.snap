exports[`<Highlights> matches snapshot 1`] = `
<div
  class="highlights-container row">
  <div
    class="price-block">
    <div
      class="row">
      <div
        class="standard-deviation-block">
        <h5
          class="standard-deviation-title">
          Std deviation -1
        </h5>
        <h5
          class="sd-highlight">
          $1
        </h5>
      </div>
      <div
        class="avg-price-block">
        <h5
          class="avg-price-title">
          Average
        </h5>
        <h5
          class="avg-price-highlight">
          $2
        </h5>
      </div>
      <div
        class="standard-deviation-block">
        <h5
          class="standard-deviation-title">
          Std deviation +1
        </h5>
        <h5
          class="sd-highlight">
          $3
        </h5>
      </div>
    </div>
  </div>
  <div
    class="proposed-price-block"
    style="display:block;">
    <h5
      class="proposed-price-title">
      Proposed price
    </h5>
    <h5
      class="proposed-price-highlight">
      $5
    </h5>
  </div>
</div>
`;
